tools/generators/workspace-generator/index.ts

```diff
@@ -0,0 +1,50 @@
+import * as path from 'path';
+import { Tree, formatFiles, generateFiles, names, offsetFromRoot } from '@nrwl/devkit';
+
+import { WorkspaceGeneratorGeneratorSchema } from './schema';
+
+interface NormalizedSchema extends ReturnType<typeof normalizeOptions> {}
+
+export default async function(host: Tree, schema: WorkspaceGeneratorGeneratorSchema) {
+  const options = normalizeOptions(host, schema);
+
+  addFiles(host, options);
+
+  if (!options.skipFormat) {
+    await formatFiles(host);
+  }
+}
+
+function normalizeOptions(host: Tree, options: WorkspaceGeneratorGeneratorSchema) {
+  if (options.name.length === 0) {
+    throw new Error('name is required');
+  }
+
+  const defaults = { skipFormat: false };
```

Comment:
this default value is declared in `schema.json`, does this mean `nx` does not handle default values declared in the schema ?

Reply:
> this default value is declared in schema.json, does this mean nx does not handle default values declared in the schema ?

- it does, typescript doesn't - the ultimate problem of default vs optional values.

JSX has special hack for that in TS compiler (via `*.defaultProps`), outside of that we are out of luck. I mean it can be hacked via additional TS type transforms which adds additional churn, from my experience this is the most concise way. If you have other proposal how to handle this lemme know!